trickster/endpoints/mocked.py

```diff
@@ -17,10 +17,22 @@
 @router.api_route('/{path:path}', methods=http.HTTPMethod)  # type: ignore
 def mocked_response(request: Request, mocked_router: Router = Depends(get_router)) -> JSONResponse:
     """All-catching route that mocks client service."""
+    response = None
+
     if match := mocked_router.match(request):  # noqa: SIM102 - nested if statements
-        if response := match.route.get_response(match):
-            match.route.hits += 1
-            response.hits += 1
-            response.delay_response()
-            return response.as_fastapi_response()
-    raise HTTPException(status_code=404, detail='No route or response was found for your request.')
+        try:
+            match.route.authenticate(request)
+            if matched_response := match.route.get_response(match):
+                match.route.hits += 1
+                response = matched_response
+        except HTTPException:
```

Comment:
This is simply wrong. How do you know that every HTTPException means 401? Don't do this. Raise some custom AuthError and translate it here. Don't use HTTPException in router or other models